engines/mohawk/myst_state.cpp

```diff
@@ -41,12 +41,15 @@ MystSaveMetadata::MystSaveMetadata() {
 	saveHour = 0;
 	saveMinute = 0;
 	totalPlayTime = 0;
+	version = 2;
+	autoSave = 0;
 }
 
 bool MystSaveMetadata::sync(Common::Serializer &s) {
```

Comment:
This should be something like that:

```java
	if (!s.syncVersion(kCurrentVersion)) {
 		return false;
 	}
 
 	s.syncAsByte(saveDay);
 	s.syncAsByte(saveMonth);
 	s.syncAsUint16LE(saveYear);
 	s.syncAsByte(saveHour);
 	s.syncAsByte(saveMinute);
 	s.syncString(saveDescription);
 	s.syncAsUint32LE(totalPlayTime);
 	s.syncAsByte(autoSave, 2);
```

The serializer methods take the version number since when to read the data as an optional parameter. With this code, if `autoSave` is initialized with the proper default value, it will only be read for saves with version 2 or more.

No need to add a version field. The serializer takes care of it internally.